pkg/docker/docker.go

```diff
@@ -544,8 +545,11 @@ func runContainerTar(opts RunContainerOptions, imageMetadata *docker.Image) (cmd
 	return cmd, tarDestination
 }
 
-// this funtion simply abstracts out the running of the newly produced image when the --run=true option is provided
-func runContainerDockerRun(container *docker.Container, d *stiDocker, image string) {
+// dumpContainerInfoAndWait dumps information about a running container (port/ip/etc) and then
+// waits for the container to exit (which it may never do).  Used when --run=true is passed
```

Comment:
It is still not clear to me that it will indeed wait for the container to exit. The `select {}` statement suggests it will block forever without any relation to the container state?